src/main/scala/kubeyml/deployment/api/package.scala

```diff
@@ -33,65 +33,75 @@ package object api {
 
   implicit def fromNonEmptyString(value: NonEmptyString): String = value.value
 
-
   implicit final class KubernetesDeploymentOps(val kubernetesDeployment: EmptyDeployment) extends AnyVal {
-    def namespace(name: String): NamespaceDeployment =  NamespaceDeployment(name)
+    def namespace(name: String): NamespaceDeployment = NamespaceDeployment(name)
   }
   implicit final class NamespaceDeploymentOps(val namespaceDeployment: NamespaceDeployment) extends AnyVal {
     def service(name: String): AppDeployment = AppDeployment(namespaceDeployment.namespace, name)
   }
 
   implicit final class AppDeploymentOps(val appDeployment: AppDeployment) extends AnyVal {
+
     def withImage(dockerImage: String): DockerisedAppDeployment =
       DockerisedAppDeployment(appDeployment.namespace, appDeployment.service, dockerImage)
   }
 
   implicit final class DockerisedAppDeploymentOps(val dockerisedAppDeployment: DockerisedAppDeployment) extends AnyVal {
     import dockerisedAppDeployment._
+
     def withProbes(livenessProbe: Probe, readinessProbe: Probe): Deployment = Deployment(
-        metadata = DeploymentMetadata(name = dockerisedAppDeployment.service, namespace = namespace),
-        spec = Spec(
-          replicas = 2,
-          selector = Selector(MatchLabels(Some(service))),
-          template = Template(
-            TemplateMetadata(
-              Labels(service),
-              Map.empty
-            ),
-            TemplateSpec(
-              List(
-                Container(
-                  service,
-                  image,
-                  List.empty,
-                  Always,
-                  livenessProbe,
-                  readinessProbe,
-                  Map.empty
-                )
+      metadata = DeploymentMetadata(name = dockerisedAppDeployment.service, namespace = namespace),
+      spec = Spec(
+        replicas = 2,
+        selector = Selector(MatchLabels(Some(service))),
+        template = Template(
+          TemplateMetadata(
+            Labels(service),
+            Map.empty
+          ),
+          TemplateSpec(
+            List(
+              Container(
+                service,
+                image,
+                List.empty,
+                List.empty,
+                IfNotPresent,
```

Comment:
is there a reason for changing that? My initial thinking is I would like the Always to be safe, and IfNotPresent is an optimisation that you can change (with an additional sbt key).
I can open another ticket for this to give the option

Reply:
ticket created https://github.com/vaslabs/sbt-kubeyml/issues/9

Reply:
Reason for changing that was by default, the images we build, our immutable. So we always use "IfNotPresent". Also during development/testing phase in local, I would prefer to use the locally image build, over the published image in docker repo.

We can have that option of choosing the defaults.

Reply:
Ok, I have made the changes to make it configurable anyway, please remember to update the readme too